src/huggingface_hub/utils/_errors.py

```diff
@@ -71,131 +132,183 @@ class LocalEntryNotFoundError(EntryNotFoundError, FileNotFoundError, ValueError)
-        e.args = (e.args[0] + f" (Request ID: {request_id})",) + e.args[1:]
-
 
-def _add_server_message_to_error_args(e: HTTPError, response: Response):
+def hf_raise_for_status(
+    response: Response, endpoint_name: Optional[str] = None
+) -> None:
     """
-    If the server response raises an HTTPError, we try to decode the response body and
-    find an error message. If the server returned one, it is added to the HTTPError
-    message.
-    """
-    try:
-        server_message = response.json().get("error", None)
-    except JSONDecodeError:
-        return
+    Internal version of `response.raise_for_status()` that will refine a
+    potential HTTPError. Raised exception will be an instance of `HfHubHTTPError`.
 
-    if (
-        server_message is not None
-        and len(server_message) > 0
-        and len(e.args) > 0
-        and isinstance(e.args[0], str)
-    ):
-        e.args = (e.args[0] + "\n\n" + str(server_message),) + e.args[1:]
+    This helper is meant to be the unique method to raise_for_status when making a call
+    to the Hugging Face Hub.
 
+    Usage:
+    ```py
+        import requests
+        from huggingface_hub.utils import hf_raise_for_status, HfHubHTTPError
 
-def _raise_for_status(response):
-    """
-    Internal version of `response.raise_for_status()` that will refine a
-    potential HTTPError.
+        response = requests.post(...)
+        try:
+            hf_raise_for_status(response)
+        except HfHubHTTPError as e:
+            print(str(e)) # formatted message
+            e.request_id, e.server_message # details returned by server
+
+            # Complete the error message with additional information once it's raised
+            e.append_to_message("\n`create_commit` expects the repository to exist.")
+            raise
+    ```
+
+    Args:
+        response (`Response`):
+            Response from the server.
+        endpoint_name (`str`, *optional*):
+            Name of the endpoint that has been called. If provided, the error message
+            will be more complete.
+
+    <Tip warning={true}>
+
+    Raises when the request has failed:
+
+        - [`~huggingface_hub.utils.RepositoryNotFoundError`]
+            If the repository to download from cannot be found. This may be because it
+            doesn't exist, because `repo_type` is not set correctly, or because the repo
+            is `private` and you do not have access.
+        - [`~huggingface_hub.utils.RevisionNotFoundError`]
+            If the repository exists but the revision couldn't be find.
+        - [`~huggingface_hub.utils.EntryNotFoundError`]
+            If the repository exists but the entry (e.g. the requested file) couldn't be
+            find.
+        - [`~huggingface_hub.utils.BadRequestError`]
+            If request failed with a HTTP 400 BadRequest error.
+        - [`~huggingface_hub.utils.HfHubHTTPError`]
+            If request failed for a reason not listed above.
```

Comment:
```suggestion
        - [`~utils.RepositoryNotFoundError`]
            If the repository to download from cannot be found. This may be because it
            doesn't exist, because `repo_type` is not set correctly, or because the repo
            is `private` and you do not have access.
        - [`~utils.RevisionNotFoundError`]
            If the repository exists but the revision couldn't be find.
        - [`~utils.EntryNotFoundError`]
            If the repository exists but the entry (e.g. the requested file) couldn't be
            find.
        - [`~utils.BadRequestError`]
            If request failed with a HTTP 400 BadRequest error.
        - [`~utils.HfHubHTTPError`]
            If request failed for a reason not listed above.
```